hpcc-nagios-tools/hpcc_centralized_nagios.py

```diff
@@ -0,0 +1,707 @@
+        doc_root = ParseXML.getroot()
+        for i in doc_root.findall('.//DaliServerProcess/Instance'):
+            DaliIP = i.attrib['netAddress']
+        return DaliIP
+
+    def eclccservers(self):
+        ParseXML = ET.parse(xml_local)
+        doc_root = ParseXML.getroot()
+        eclservers = []
+        for i in doc_root.findall('.//EclCCServerProcess/Instance'):
+            ECLIP = i.attrib['netAddress']
+            eclservers.append(ECLIP)
+        return eclservers
+
+    def eclservers(self):
+        ParseXML = ET.parse(xml_local)
+        doc_root = ParseXML.getroot()
+        eclservers = []
+        for i in doc_root.findall('.//EclServerProcess/Instance'):
+            ECLIP = i.attrib['netAddress']
+            eclservers.append(ECLIP)
+        return eclservers
+
+    def eclagent(self):
+        ParseXML = ET.parse(xml_local)
+        doc_root = ParseXML.getroot()
+        agents = []
+        for i in doc_root.findall('.//EclAgentProcess/Instance'):
+            AgentIP = i.attrib['netAddress']
+            agents.append(AgentIP)
+        return agents
+
+    def dfuserver(self):
+        ParseXML = ET.parse(xml_local)
+        doc_root = ParseXML.getroot()
+        dfuserver = []
+        for i in doc_root.findall('.//DfuServerProcess/Instance'):
+            DfuServer = i.attrib['netAddress']
+            dfuserver.append(DfuServer)
+        return dfuserver
+
+    def ESP(self):
+        ParseXML = ET.parse(xml_local)
+        doc_root = ParseXML.getroot()
+        ESP = []
+        for i in doc_root.findall('.//EspProcess/Instance'):
+            EspServer = i.attrib['netAddress']
+            ESP.append(EspServer)
+        return ESP
+
+    def EclScheduler(self):
+        ParseXML = ET.parse(xml_local)
+        doc_root = ParseXML.getroot()
+        eclscheduler = []
+        for i in doc_root.findall('.//EclSchedulerProcess/Instance'):
+            scheduler = i.attrib['netAddress']
+            eclscheduler.append(scheduler)
+        return eclscheduler
+
+    def sasha(self):
+        ParseXML = ET.parse(xml_local)
+        doc_root = ParseXML.getroot()
+        for i in doc_root.findall('.//SashaServerProcess/Instance'):
+            Sasha = i.attrib['netAddress']
+        return Sasha
+
+    def Roxie(self):
+        ParseXML = ET.parse(xml_local)
+        doc_root = ParseXML.getroot()
+        Roxies = []
+        for roxie in doc_root.findall('.//RoxieServerProcess'):
+            Roxie = roxie.attrib['computer']
+            Roxies.append(Roxie)
+        ip_addr = {}
+        for node in doc_root.findall('.//Hardware/Computer'):
+            Machine = node.attrib['name']
+            IP = node.attrib['netAddress']
+            ip_addr[Machine] = IP
+        RoxieSlaves = []
+        for key in ip_addr:
+            if key not in Roxies:
+                continue
+            elif key in Roxies:
+                RoxieSlaves.append(ip_addr[key])
+        return RoxieSlaves
```

Comment:
Roxie, Thor Master, Thor Slaves and MySQLRepo can be grouped to another type of function with process name as input parameter.